src/Compilers/Core/Portable/DiagnosticAnalyzer/DefaultAnalyzerAssemblyLoader.Desktop.cs

```diff
@@ -102,106 +69,27 @@ internal bool EnsureResolvedUnhooked()
             return false;
         }
 
-        private Assembly? CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
+        private Assembly? AssemblyResolve(object sender, ResolveEventArgs args)
         {
-            // In the .NET Framework, if a handler to AssemblyResolve throws an exception, other handlers
-            // are not called. To avoid any bug in our handler breaking other handlers running in the same process
-            // we catch exceptions here. We do not expect exceptions to be thrown though.
             try
             {
```

Comment:
Consider wrapping `try { ... } catch { ... }` around `return Assembly.LoadFrom(bestPath);` only, assuming that is the only method expected to throw an exception.

Reply:
`ApplyPolicy`, `new AssemblyName` and `LoadFrom` can all throw. The first two are significantly more edge case than the last one but they can indeed throw. Basically only `GetBestPath` is a non-throwing API.